deafrica/data/gmw.py

```diff
@@ -0,0 +1,152 @@
+import json
+import os
+import sys
+from datetime import datetime
+from subprocess import check_output, STDOUT, CalledProcessError
+
+import click
+import pystac
+from odc.aws import s3_dump
+from odc.index import odc_uuid
+from pystac import Item
+from pystac.utils import datetime_to_str
+from rio_stac import create_stac_item
+from urlpath import URL
+
+from deafrica.utils import setup_logging
+
+SOURCE_URL_PATH = URL(f"https://wcmc.io/")
+FILE_NAME = "GMW_{year}"
+
+# Set log level to info
+log = setup_logging()
+
+
+def download_and_unzip_gmw(year):
+    import requests
+    import shutil
+    from zipfile import ZipFile
+
+    local_filename = FILE_NAME.format(year=year)
+    url = SOURCE_URL_PATH / local_filename
+
+    with requests.get(url, stream=True, allow_redirects=True) as r:
+        with open(local_filename, "wb") as f:
+            shutil.copyfileobj(r.raw, f)
+
+    with ZipFile(local_filename) as z:
+        z.extractall()
+
+    return [f for f in z.namelist() if f.endswith(".shp")][0]
+
+
+def create_item_and_dump(cog_file: str, s3_dst: str, year) -> Item:
+    out_path = URL(f"{s3_dst}/{year}/")
+    file_name = cog_file.replace("tif", "")
+
+    log.info("Item base creation")
+    item = create_stac_item(
+        file_name,
+        id=str(odc_uuid("chirps", "2.0", [file_name])),
+        with_proj=True,
+        input_datetime=datetime(int(year), 12, 31),
+        properties={
+            "odc:processing_datetime": datetime_to_str(datetime.now()),
+            "odc:product": "rainfall_chirps_monthly",
+            "start_datetime": f"{year}-01-01T00:00:00Z",
+            "end_datetime": f"{year}-12-31T23:59:59Z",
+        },
+    )
+
+    log.info("links creation")
+    item.set_self_href(out_path / f"gmw_{year}_stac-item.json")
+    item.add_links(
+        [
+            pystac.Link(
+                target=SOURCE_URL_PATH / FILE_NAME.format(year=year),
+                title="Source file",
+                rel=pystac.RelType.DERIVED_FROM,
+                media_type="application/zip",
+            )
+        ]
+    )
+
+    log.info("assets creation")
+    del item.assets["asset"]
+
+    item.assets["rainfall"] = pystac.Asset(
+        href=out_path / file_name,
+        title="gmw-v1.0",
+        media_type=pystac.MediaType.COG,
+        roles=["data"],
+    )
+
+    log.info(f"Item created {item.to_dict()}")
+
+    log.info("Dump the data to S3")
+    out_data = out_path / cog_file
+    s3_dump(file_name, out_data, ACL="bucket-owner-full-control")
+    log.info(f"File written to {out_data}")
+
+    log.info("Write STAC to S3")
+    s3_dump(
+        json.dumps(item.to_dict(), indent=2),
+        item.self_href,
+        ContentType="application/json",
+        ACL="bucket-owner-full-control",
+    )
+    log.info(f"STAC written to {item.self_href}")
+
+
+def download_gmw(year: str, s3_dst: str, crs="EPSG:6933", res=10):
```

Comment:
this function does more than just `download_gmw` so please think better naming

Reply:
Done